Buildings/Controls/OBC/ASHRAE/FanCoilUnit/Subsequences/SupplyAirTemperature.mo

```diff
@@ -0,0 +1,514 @@
+      enable = controllerTypeHeaCoi == Buildings.Controls.OBC.CDL.Types.SimpleController.PI or
+      controllerTypeHeaCoi == Buildings.Controls.OBC.CDL.Types.SimpleController.PID));
+
+  parameter Real TdHeaCoi(
+    final unit="s",
+    displayUnit="s",
+    final quantity="time")=0.1
+    "Derivative block time constant"
+    annotation(Dialog(tab="PID controller parameters", group="Heating coil",
+      enable = controllerTypeHeaCoi == Buildings.Controls.OBC.CDL.Types.SimpleController.PD or
+      controllerTypeHeaCoi == Buildings.Controls.OBC.CDL.Types.SimpleController.PID));
+
+  parameter Real deaHysLim(
+    final unit="1",
+    displayUnit="1") = 0.01
+    "Hysteresis limits for deadband mode transitions"
+    annotation(Dialog(tab="Advanced"));
+
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanInput u1Fan
+    "Fan proven on signal"
+    annotation (Placement(transformation(extent={{-160,120},{-120,160}}),
+      iconTransformation(extent={{-140,80},{-100,120}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput uHea(
+    final unit="1",
+    displayUnit="1") if have_heaCoi
+    "Heating loop signal"
+    annotation (Placement(transformation(extent={{-160,20},{-120,60}}),
+      iconTransformation(extent={{-140,0},{-100,40}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput uCoo(
+    final unit="1",
+    displayUnit="1") if have_cooCoi
+    "Cooling loop signal"
+    annotation (Placement(transformation(extent={{-160,-60},{-120,-20}}),
+      iconTransformation(extent={{-140,-80},{-100,-40}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput TAirSup(
+    final unit="K",
+    displayUnit="K",
+    final quantity="ThermodynamicTemperature")
+    "Measured supply air temperature"
+    annotation (Placement(transformation(extent={{-160,-20},{-120,20}}),
+      iconTransformation(extent={{-140,-40},{-100,0}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput TZonHeaSet(
+    final unit="K",
+    displayUnit="K",
+    final quantity="ThermodynamicTemperature") if have_heaCoi
+    "Zone heating temperature setpoint"
+    annotation (Placement(transformation(extent={{-160,60},{-120,100}}),
+      iconTransformation(extent={{-140,40},{-100,80}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput TZonCooSet(
+    final unit="K",
+    displayUnit="K",
+    final quantity="ThermodynamicTemperature") if have_cooCoi
+    "Zone cooling temperature setpoint"
+    annotation (Placement(transformation(extent={{-160,-140},{-120,-100}}),
+      iconTransformation(extent={{-140,-120},{-100,-80}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealOutput yHeaCoi(
+    final unit="1",
+    displayUnit="1") if have_heaCoi
+    "Heating coil signal"
+    annotation (Placement(transformation(extent={{120,40},{160,80}}),
+      iconTransformation(extent={{100,40},{140,80}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealOutput yCooCoi(
+    final unit="1",
+    displayUnit="1") if have_cooCoi
```

Comment:
The `yCooCoi` and `yHeaCoi` are conditional connectors. However, the conditions are not propagated upstream. For example, the instances `mul3`, `swiCooCoi`, `cooPIDCoo` can be conditional removed if `have_cooCoi=false`. so that the parameters needed in instance `cooPIDCoo` can be disabled.

The conditions to diable parameters may need to be propagated to `Buildings.Controls.OBC.ASHRAE.FanCoilUnit.Controller`.